### PR TITLE
Autosurgeons can now be used on other people

### PR DESCRIPTION
**Pull request by Zonespace27**
## About The Pull Request
Autosurgeons can now be used on people other than yourself, after an 8 second actionbar (multiplied by 3/4ths for syndicate ones).

## Why It's Good For The Game
Felt like it might be interesting for the game, seemed a bit odd that it couldn't be done.

## Changelog

:cl:
qol: You can now use autosurgeons on other people after a delay.
/:cl:

### CODE REVIEW

File: code/modules/surgery/organs/autosurgeon.dm [hunk 1]
```diff
@@ -69,6 +71,20 @@
 	else
 		return ..()
 
+/obj/item/autosurgeon/organ/attack(mob/living/target, mob/living/user, params)
+	add_fingerprint(user)
+	to_chat(user, span_notice("You begin to prepare to use [src] on [target]."))
+	if(do_after(user, (8 SECONDS * surgery_speed), target))
```
Comment (Mothblocks) [inline code/modules/surgery/organs/autosurgeon.dm:77]: Early-return

File: code/modules/surgery/organs/autosurgeon.dm [hunk 2]
```diff
@@ -69,6 +71,20 @@
 	else
 		return ..()
 
+/obj/item/autosurgeon/organ/attack(mob/living/target, mob/living/user, params)
+	add_fingerprint(user)
+	to_chat(user, span_notice("You begin to prepare to use [src] on [target]."))
```
Comment (Mothblocks) [inline code/modules/surgery/organs/autosurgeon.dm:76]: Replace this with a `balloon_alert_to_viewers`--the target should know they're being operated on

File: code/modules/surgery/organs/autosurgeon.dm [hunk 3]
```diff
@@ -69,6 +71,20 @@
...
+	if(do_after(user, (8 SECONDS * surgery_speed), target))
+		user.visible_message(span_notice("[user] presses a button on [src], and you hear a short mechanical noise."), span_notice("You press a button on [src] as it plunges into [target]'s body."))
+		to_chat(target, span_notice("You feel a sharp sting as something plunges into your body!"))
+		playsound(get_turf(user), 'sound/weapons/circsawhit.ogg', 50, TRUE)
```
Comment (Mothblocks) [inline code/modules/surgery/organs/autosurgeon.dm:80]:
```suggestion
		playsound(get_turf(user), 'sound/weapons/circsawhit.ogg', 50, vary = TRUE)
```

File: code/modules/surgery/organs/autosurgeon.dm [hunk 4]
```diff
@@ -69,6 +71,20 @@
...
+		storedorgan = null
+		name = initial(name)
+		if(uses != INFINITE)
+			uses--
+		if(!uses)
+			desc = "[initial(desc)] Looks like it's been used up."
```
Comment (Mothblocks) [inline code/modules/surgery/organs/autosurgeon.dm:86]: This is copied and pasted, please move this to a shared proc

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by jughu** [comment]
But this wont see practice besides admin stuff/nuke op autosurgeons

**Comment by Mothblocks** [comment]
Thats fine its neat you can do it

**Pull request merged**